lib/gui/components/drive-selector/controllers/drive-selector.js

```diff
@@ -75,20 +90,38 @@ module.exports = function (
       return $q.resolve(false)
     }
 
-    if (constraints.isDriveSizeRecommended(drive, selectionState.getImage())) {
-      return $q.resolve(true)
+    const isLargeDrive = constraints.isDriveSizeLarge(drive)
+    const isRecommendedDrive = constraints.isDriveSizeRecommended(drive, selectionState.getImage())
+
+    if (isLargeDrive || !isRecommendedDrive) {
+      const messageTuples = [
+        [
+          isLargeDrive,
+          messages.warning.largeDriveSize({
+            humanDriveSize: units.bytesToClosestUnit(drive.size)
+          })
+        ],
+        [
+          !isRecommendedDrive,
+          messages.warning.unrecommendedDriveSize({
+            drive,
+            image: selectionState.getImage()
+          })
+        ]
+      ]
+
+      const concatenatedMessage = _.reduce(messageTuples, (accumulatedMessage, messageTuple) => {
+          if (messageTuple[0]) {
+            return `${accumulatedMessage}\n\n${messageTuple[1]}`
+          } else {
+            return accumulatedMessage
+          }
+        }, '')
```

Comment:
Why does this need to use `reduce` rather than just a straightforward `join`?

Reply:
I guess we could `filter` and then `join` but that's two loops, but the reason we use `reduce` here is so we can pick which strings to add to the dialog in one recursion over the list rather than two since JavaScript doesn't do stream-fusion. This doesn't matter anymore now that we're not using this though!

Reply:
Ahhh, I'd misunderstood what `messageTuples` actually contains - didn't realise it contained a boolean condition flag in addition to the message itself.